CommandScreen.c

```diff
@@ -49,21 +49,27 @@ static int CommandScreen_scanAscii(InfoScreen* this, const char* p, size_t total
 static int CommandScreen_scanWide(InfoScreen* this, const char* p, size_t total, char* line) {
    mbstate_t state;
    memset(&state, 0, sizeof(state));
-   int line_cols = 0;
-   int line_offset = 0, line_size = 0, last_spc_cols = -1, last_spc_offset = -1;
+   size_t bytes;
+   int line_cols = 0, line_offset = 0, line_size = 0, width = 0;
+   int last_spc_cols = -1, last_spc_offset = -1;
    for (size_t i = 0; i < total; ) {
       assert(line_offset >= 0 && (size_t)line_offset <= total);
-      wchar_t wc;
-      size_t bytes = mbrtowc(&wc, p + i, total - i, &state);
-      int width = wcwidth(wc);
-      if (bytes == (size_t)-1 || bytes == (size_t)-2 || bytes == 0 || width < 0) {
+      unsigned char c = (unsigned char)p[i];
+      if (c < 0x80) { // skip mbrtowc for ASCII characters
+         if (c == ' ') {
+            last_spc_offset = line_offset;
+            last_spc_cols = line_cols;
+         }
          bytes = width = 1;
-      }
-
-      memcpy(line + line_offset, p + i, bytes);
-      if (wc == ' ') {
-         last_spc_offset = line_offset;
-         last_spc_cols = line_cols;
+         line[line_offset] = c;
+      } else {
+         wchar_t wc;
+         bytes = mbrtowc(&wc, p + i, total - i, &state);
+         width = wcwidth(wc);
```

Comment:
Check `bytes` before calling `wcwidth`. If `mbrtowc` returns an error, `wc` can be left undefined.